packages/web-components/fast-foundation/src/popover/popover.ts

```diff
@@ -0,0 +1,634 @@
+import { attr, DOM, FASTElement, observable } from "@microsoft/fast-element";
+import { Direction, keyCodeEscape, keyCodeTab } from "@microsoft/fast-web-utilities";
+import tabbable from "tabbable";
+import { AnchoredRegion, AxisPositioningMode, AxisScalingMode } from "../anchored-region";
+import { ARIAGlobalStatesAndProperties } from "../patterns";
+import { applyMixins, getDirection } from "../utilities";
+import { PopoverPosition } from "./popover.options";
+
+// TODO: ASK Do we want trapFocus to always be true???
+// TODO: ASK Should we bring focus back to target after popover is closed
+
+export { PopoverPosition };
+
+/**
+ * An Popover Custom HTML Element.
+ *
+ * @public
+ */
+export class Popover extends FASTElement {
+    private static DirectionAttributeName: string = "dir";
```

Comment:
I believe this is unused.

Reply:
Likely, I haven't finished this just yet, but will take a look at this.